airflow/providers/cncf/kubernetes/utils/xcom_sidecar.py

```diff
@@ -0,0 +1,60 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+This module provides an interface between the previous Pod
+API and outputs a kubernetes.client.models.V1Pod.
+The advantage being that the full Kubernetes API
+is supported and no serialization need be written.
+"""
+import copy
+
+from kubernetes.client import models as k8s
+
+MAX_LABEL_LEN = 63
+
+
+class PodDefaults:
```

Comment:
Does this mean we have 2/3 PodDefaults:

1) there
2) https://github.com/apache/airflow/blob/adb505d0f9414f5b604c3f5505cd908ef5f77f96/airflow/kubernetes/pod_generator.py#L45-L63

and 

3) Deprecated module:

https://github.com/apache/airflow/blob/adb505d0f9414f5b604c3f5505cd908ef5f77f96/airflow/kubernetes/pod_generator_deprecated.py#L36-L54

Reply:
@kaxil I would love some guidance on how to start deleting or deprecating. the pod_generator_deprecated was brought over for certain backcompat functions from 1.10.  I think we could potentially delete the one in pod_generator and import from pod_generator_deprecated?

Reply:
Yeah we can only deprecared module in 3.0 unfortunately but your last commit should remove (2).

Add a note in Updating.md too i.e if someone was relying on PodDefaults use one from `airflow/providers/cncf/kubernetes/utils/xcom_sidecar.py`.

And then we can chuck that out in 3.0

Reply:
@kaxil sounds good